### ISSUE TITLE
Selecting the closest-matching locale

### ISSUE THREAD

**Issue by lornemcintosh** [yahoo/ember-intl#274]
I'm setting the locale in my application like this:
```javascript
this.get('intl').setLocale(navigator.languages);
```
In Firefox, `navigator.languages` supplies an array of the user's preferred languages in order from most to least preferred. For example, `['en-CA', 'en']`. However, if the only translation in my application is en-US, then the user gets the "Missing translation" error on every string. Obviously an English-speaking Canadian would accept the en-US locale, or (regardless of his patriotism) would at least prefer this over nothing. Thus I'd like the en-US locale to be used in this situation.

What is the best practice for setting locale? Does ember-intl help at all, or am I left to devise my own rules/heuristics? I've set 'en-US' as my "default locale",
```javascript
intl: {
   defaultLocale: 'en-us'
},
```
but that doesn't seem to have any effect in this regard.

**Comment by jasonmit** [comment]
While I work through the best way to handle this, the following should work:

```js
const locales = Ember.A(navigator.languages.concat(['en-US'])).uniq();
this.get('intl').setLocale(locales);
```

Keeping the issue opened though.

**Comment by lornemcintosh** [comment]
FYI, I found this RFC which seems to sort-of address this. I didn't know too much about the topic, so I found it interesting: https://tools.ietf.org/html/rfc4647

**Comment by lornemcintosh** [comment]
As a prerequisite to tackling this, how can I get a list of the supported locales from ember-intl? If nothing else, I could at least use this to create a dropdown for the user to select their preferred locale.

**Comment by jasonmit** [comment]
You want to display a list of all the locales that ember-intl supports or that your application has translations for?

**Comment by machty** [comment]
I ran into this same issue. It's worth mentioning that the [newest Android has introduced APIs for selecting the best match of locales](https://developer.android.com/guide/topics/resources/multilingual-support.html), which might be a nice pattern to adopt.

One thing that was a bit tricky was that we also wanted the ability to default all English or Spanish locales to the single respective set of English / Spanish translations that we supported, while preserving the localized date/time/number formatting of the user's preferred locale. In other words, we only had translations for `es`, so we wanted users from both Mexico and Spain to use the same translation set, but to format the numbers/dates/currencies according to their base locale. 

There didn't seem to be a built in way to support this, so I ended up stashing the "preferred" locale, and overriding the ember-intl service to always provide that locale for `formatNumber`/`formatDate`/`formatTime`.
